src/Symfony/Bridge/Twig/AppVariable.php

```diff
@@ -93,7 +93,7 @@ public function getRequest(): ?Request
     /**
      * Returns the current session.
      */
-    public function getSession(): ?Session
+    public function getSession(): ?FlashBagAwareSessionInterface
```

Comment:
The return type could be the intersection of all the interfaces implemented by `Session` in `symfony/http-foundation: 5.4`.

```suggestion
    public function getSession(): ?SessionInterface&\Traversable&\Countable
```

Reply:
@GromNaN Native intersection types are not nullable. And unions of a intersection type with the `null` type require PHP >= 8.2. That aside, your proposal generates the same problem as the current type: The HttpFoundation `Request` cannot guarantee that the session it returns in `Traversable` and `Countable`.

Reply:
Well, I've forgotten nullable in my proposal. Clearly, the return type cannot be changed in a totally backwards-compatible way.

Noticed [your proposition](https://github.com/symfony/symfony/pull/50794#issuecomment-1611409156) looks more correct.